php/class-photonfill.php

```diff
@@ -779,17 +779,12 @@ public function get_breakpoint_url( $attachment_id, $img_size, $breakpoint, $pix
 
 		/**
 		 * Swap `lazyloaded` class for `lazyload`. Why is this necessary?
-		 * We need to unveil lazyload images in TinyMCE, which removes the `lazyload` class and adds the `lazyloaded` class. This means on the front end
-		 * lazysizes won't pick up on the image b/c it assumes the image has already been lazyloaded.
-		 * @param int $post_id
+		 * We need to unveil lazyload images in TinyMCE, which removes the `lazyload` class and adds the `lazyloaded` class.
+		 * This means on the front end lazysizes won't pick up on the image b/c it assumes the image has already been lazyloaded.
+		 * @param string $content
 		 */
-		public function swap_lazyload_classes( $post_id ) {
-			$current_post = get_post( $post_id );
-			$current_post->post_content = preg_replace( '/(class="[^"]*)(lazyloaded)([^"]*")/i', '$1' . 'lazyload' . '$3', $current_post->post_content );
-
-			remove_action( 'save_post', array( $this, 'swap_lazyload_classes' ) );
-			wp_update_post( $current_post );
-			add_action( 'save_post', array( $this, 'swap_lazyload_classes' ) );
+		public function swap_lazyload_classes( $content ) {
+			return preg_replace( '/(class=\\\\"[^"]*)(lazyloaded)([^"]*")/i', '$1' . 'lazyload' . '$3', $content );
```

Comment:
the content contains _escaped_ text at this stage... (so double quotes are being escaped with a `\` at this point in preparation for mysql).

in a regex you need to escape a backslash with another literal backslash (because a backslash is an escape character).

and even more confusing, `preg_replace` will escape the escape characters in the string before evaluating the replacement http://stackoverflow.com/questions/2222643/php-preg-replace

so `\\\\` gets translated to `\\` by `preg_replace` which gets translated to `\` by the regex. 😄 

> regex is really great to work with
- said no one ever